dask_snowflake/core.py

```diff
@@ -140,12 +140,60 @@ def _fetch_query_batches(query, connection_kwargs, execute_params):
     return batches
 
 
+def _partition_batches(
+    batches: list[ArrowResultBatch],
+    meta: pd.DataFrame,
+    npartitions: None | int = None,
+    partition_size: None | str | int = None,
+) -> list[list[ArrowResultBatch]]:
+    """
+    Given a list of batches and a sample, partition the batches into dask dataframe
+    partitions.
+
+    Batch sizing is seemingly not under our control, and is typically much smaller
+    than the optimal partition size:
+    https://docs.snowflake.com/en/user-guide/python-connector-distributed-fetch.html
+    So instead batch the batches into partitions of approximately the right size.
+    """
+    if npartitions is not None:
+        assert npartitions >= 1
+        target = sum([b.rowcount for b in batches]) // npartitions
+    elif partition_size is not None:
+        partition_bytes = (
+            parse_bytes(partition_size)
+            if isinstance(partition_size, str)
+            else partition_size
+        )
+        approx_row_size = meta.memory_usage().sum() / len(meta)
+        target = max(partition_bytes / approx_row_size, 1)
+    else:
+        raise ValueError("Must provide one of `npartitions` or `partition_size`")
+
+    batches_partitioned: list[list[ArrowResultBatch]] = []
+    curr: list[ArrowResultBatch] = []
+    partition_len = 0
+    for batch in batches:
+        if len(curr) > 0 and batch.rowcount + partition_len > target:
```

Comment:
So we're implicitly making the assumption that rows are roughly equal size, yes?  Totally fine, just making sure I understand.

Reply:
Yes, I'd go so far as to say I'm explicitly making that assumption :) This isn't a very sophisticated bit of logic.

Reply:
Woot.